### ISSUE TITLE
Snippet preview toggle buttons accessibility improvements

### ISSUE THREAD

**Issue by afercia** [Yoast/YoastSEO.js#1047]
I'd like to propose three a11y improvements for the new snippet preview toggle buttons.

1
Buttons placement:
the new buttons are on the right of the Edit snippet button so the current logical order is:
"the Snippet preview box with details" > "Edit snippet" > "Mobile preview" > "Desktop preview"
the a11y issue here is that the "Edit snippet" button expands the editor below and, when activated, its `aria-expanded` attribute announces a change of state, giving users the indication that something in the page is now expanded.
For this reason, the expanded panel should immediately follow the button in the source order. Imagine you can't see and:
- you activate the button
- you hear "expanded" announced by your screen reader
- and then what? you have no idea what exactly is now expanded in the page
- the most natural action is going forward with exploring content, hoping to find the "expanded thing"
- instead, the first thing you find are the toggle buttons

Having the "expanded thing" right after the control that was activated solves this issue and ideally should be a pattern to always use in similar cases.
For reference, worth nothing ARIA has also an [aria-controls attribute](https://www.w3.org/TR/wai-aria/states_and_properties#aria-controls) but support is very limited (as far as I know, only JAWS with _some browsers_). Also, the only thing it does is offering users a shortcut to "jump" to the controlled element, so it's not a great option.
For more details about `aria-controls` and the reasons why we should not use it, see this post (notice the very meaningful post title):

Aria-Controls is Poop
http://www.heydonworks.com/article/aria-controls-is-poop

Changing the buttons order would result in a new logical order:
"the Snippet preview box with details" > "Mobile preview" > "Desktop preview" > "Edit snippet"
and visually would be something like this:

<img width="771" alt="screen shot 2017-01-17 at 09 30 25" src="https://cloud.githubusercontent.com/assets/1682452/22146182/10198e22-df04-11e6-96ee-b26b6d12f9f0.png">

2
There is no audible feedback about _which_ of the buttons is currently active. This can be easily solved using an `aria-pressed` attribute which would make screen readers announce the buttons as "toggle buttons" and also their selected state:

![screen shot 2017-01-20 at 11 39 18](https://cloud.githubusercontent.com/assets/1682452/22146540/46b8e45e-df05-11e6-9391-f7dc0d9cbc6c.png)

3
Lastly, I'd consider to add tooltips to the buttons. They're icon-only controls and tooltips could help users better understand their purpose.

**Comment by afercia** [comment]
About the visual change with the toggle buttons on the left, /cc @moorscode @hedgefield

**Comment by hedgefield** [comment]
Personally, I wouldn't mind this change. It puts a bit more attention weight towards those toggle buttons, which feel somehow understated when they are on the right.

I'm not sure adding text or tooltips to them would be necessary, that might obfuscate the edit button too much (though I don't know if there already is a pattern for tooltips within wordpress or our plugin?). However, in one of our brainstorms, Omar and I considered the idea of docking those buttons to the preview itself, essentially making them into tabs. We discarded the idea then, but maybe there is something to it if it helps with understanding (would require more design).

**Comment by afercia** [comment]
Thanks Tim. Re: tooltips, no there's no established pattern in WordPress. On Yoast SEO we're experimenting them since a few months, see for example the metabox left sidebar icons. Other big applications, e.g. Gmail or Github, use tooltips for controls that have only icons because icons don't have a universal meaning and need to be clarified. Other concerns come into play too, for example cognitive impairments, attention deficit, and so on. Some usability experts warn that [icons should always have some visible text](https://www.nngroup.com/articles/icon-usability/), but that maybe is a bit too much.
In this specific case, I'd propose to try them and see if they're an effective improvement. In the meanwhile, I went ahead and they would look like this:

<img width="720" alt="screen shot 2017-01-20 at 13 24 18" src="https://cloud.githubusercontent.com/assets/1682452/22150611/43249e7c-df1a-11e6-9517-d98eb7b71c03.png">

Note: tooltips can be positioned in 8 different positions:
- left or right
- top left, center, right
- bottom left, center, right

In this example I've used "bottom right" because in the responsive view there's not enough space on the left.

**Comment by hedgefield** [comment]
Looks good Andrea. Speaking from a design standpoint I'd say this is a perfectly fine improvement.
